SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -220,6 +247,22 @@ private static ProjectInfoAnalysisResult ProcessProjectInfoFiles(IEnumerable<Pro
                     {
                         string fxCopReport = TryGetFxCopReport(projectInfo, logger);
                         string vsCoverageReport = TryGetCodeCoverageReport(projectInfo, logger);
+                        if (projectInfo.Encoding != null)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 control flow statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS134)